posts/2021-05-14-ldap-multipart-21005.adoc

```diff
@@ -0,0 +1,225 @@
+}
+----
+
+Or if you're using Docker:
+
+[source]
+----
+FROM open-liberty
+----
+//end::run[]
+
+Or take a look at our link:{url-prefix}/downloads/[Downloads page].
+
+[link=https://stackoverflow.com/tags/open-liberty]
+image::img/blog/blog_btn_stack.svg[Ask a question on Stack Overflow, align="center"]
+
+//tag::features[]
+
+[#ldap]
+=== LDAP connection support for Kerberos authentication
+
+LDAP bind operations are used to authenticate clients (and the users or applications behind them) to the directory server. This establishes an authorization identity that is used for subsequent operations that are processed on that connection, and specifies the LDAP protocol version that the client uses. Before this update, the `LdapRegistry` element supported binding either anonymously or by using simple authentication with a user (`bindDN`) and password (`bindPassword`). This update adds an option to bind to LDAP: GSSAPI/Kerberos. Kerberos is an authentication mechanism that allows a client and service to mutually authenticate by a Key Distribution Center (KDC). In Open Liberty 21.0.0.5,  you can use either a Kerberos credential cache (ccache) or a Kerberos keytab file.
+
+To update an `LdapRegistry` to use the GSSAPI/Kerberos option, you can set the bind authentication mechanism type using the new `LdapRegistry` attribute, `bindAuthMechanism`:
+
+[source, xml]
+----
+bindAuthMechanism="GSSAPI"
+----
+
+You also need the Kerberos principal or Service Principal Name:
+
+[source, xml]
+----
+krb5Principal="user1@EXAMPLE.COM"
+----
+
+If you are using a Kerberos credential cache (ticket cache or ccache) to store the Kerberos credentials, add the ticket cache file name to the `LdapRegistry` with the new attribute, `krb5TicketCache`:
+
+[source, xml]
+----
+krb5TicketCache="${server.config.dir}/security/krb5-user1.cc"
+----
+
+If you are using a custom Kerberos configuration file (krb.conf or krb.ini), set the file name using the global Kerberos configuration element:
+
+[source, xml]
+----
+<kerberos configFile="${server.config.dir}/security/krb5.conf"/>
+----
+
+If you are using a Kerberos keytab file to store encrypted keys for principals, set the file using the global Kerberos configuration element:
+
+[source, xml]
+----
+<kerberos keytab="${server.config.dir}/security/krb5.keytab" configFile="${server.config.dir}/security/krb5.conf"/>
+----
+
+If the Kerberos configuration file is not defined, Open Liberty will attempt to resolve by using the JDK default locations and the operating system default locations.
+
+For the Kerberos credentials, the locations are checked in the following order: the ticket cache (if provided), the configured keytab file, and finally the JDK default location.
+
+The following example shows how to configure the `LdapRegistry` element using a ticket cache and custom Kerberos config file:
+
+[source, xml]
+----
+<kerberos keytab= configFile="${server.config.dir}/security/krb5.conf"/>
```

Comment:
@utle this looks incorrect; I'm guessing the `keytab` key needs to be removed (or otherwise provided a value)

Reply:
My bad, used an old code snippet. I believe it should be:

```
<kerberos keytab="${server.config.dir}/security/krb5.keytab" configFile="${server.config.dir}/security/krb5.conf" />
```

Reply:
@utle plz confirm the code snippet, whether the keytab should be there and if so that the value is correct

Reply:
It's looked good with Austin updated